src/java.base/share/classes/java/lang/FdLibm.java

```diff
@@ -1194,4 +1194,206 @@ static double compute(double x) {
             return y;
         }
     }
+
+    /**
+     * Method :
+     * mathematically sinh(x) if defined to be (exp(x)-exp(-x))/2
+     *      1. Replace x by |x| (sinh(-x) = -sinh(x)).
+     *      2.
+     *                                                  E + E/(E+1)
+     *          0        <= x <= 22     :  sinh(x) := --------------, E=expm1(x)
+     *                                                      2
+     *
+     *          22       <= x <= lnovft :  sinh(x) := exp(x)/2
+     *          lnovft   <= x <= ln2ovft:  sinh(x) := exp(x/2)/2 * exp(x/2)
+     *          ln2ovft  <  x           :  sinh(x) := x*shuge (overflow)
+     *
+     * Special cases:
+     *      sinh(x) is |x| if x is +INF, -INF, or NaN.
+     *      only sinh(0)=0 is exact for finite x.
+     */
+    static class Sinh {
```

Comment:
For clarity, should this not be declared `final`?

Reply:
Hmm. Wouldn't hurt to do so.

I was thinking of making a pass over Fdlibm.java at some point to make sure the nested classes had package-level rather than public accessibility and that the classes had private constructors that threw an exception (to prevent instantiation).

Reply:
It doesn't matter much. It was just a drive-by comment; I don't normally look at this part of the code but the PR title attracted my attention (it is not everyday you see hyperbolic transcendental PRs in the JDK), and when I skimmed through the code, some reflex part of my brain said "utility classes should be final".

Reply:
Changed the nested classes for the hyperbolic functions to be final and added a private constructor.